src/main/java/com/synopsys/integration/jenkins/detect/extensions/tool/FindOrInstallAirGapJar.java

```diff
@@ -0,0 +1,75 @@
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.jenkins.detect.extensions.tool;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.filefilter.PrefixFileFilter;
+
+import com.synopsys.integration.jenkins.detect.exception.DetectJenkinsException;
+import com.synopsys.integration.jenkins.detect.extensions.AirGapDownloadStrategy;
+import com.synopsys.integration.jenkins.detect.extensions.DetectDownloadStrategy;
+import com.synopsys.integration.jenkins.extensions.JenkinsIntLogger;
+import com.synopsys.integration.jenkins.service.JenkinsConfigService;
+
+public class FindOrInstallAirGapJar {
+    public static final String DETECT_JAR_PREFIX = "synopsys-detect-";
+    private final JenkinsIntLogger logger;
+    private final JenkinsConfigService jenkinsConfigService;
+
+    public FindOrInstallAirGapJar(JenkinsIntLogger logger, JenkinsConfigService jenkinsConfigService) {
+        this.logger = logger;
+        this.jenkinsConfigService = jenkinsConfigService;
+    }
+
+    public String getOrDownloadAirGapJar(DetectDownloadStrategy detectDownloadStrategy) throws IOException, InterruptedException, DetectJenkinsException {
+        Optional<DetectAirGapInstallation> detectAirGapWithName = jenkinsConfigService
+                                                                      .getInstallationForNodeAndEnvironment(DetectAirGapInstallation.DescriptorImpl.class, ((AirGapDownloadStrategy) detectDownloadStrategy).getAirGapInstallationName());
+
+        if (!detectAirGapWithName.isPresent()) {
+            throw new DetectJenkinsException(String.format("Detect AirGap cannot be executed: No DetectAirGap with the name %s could be found in the global tool configuration.", detectAirGapWithName));
+        }
+
+        return getAirGapJar(detectAirGapWithName.get());
+    }
+
+    private String getAirGapJar(DetectAirGapInstallation detectAirGapInstallation) throws DetectJenkinsException {
+        String baseDir = detectAirGapInstallation.getHome();
+
+        if (baseDir == null) {
+            throw new DetectJenkinsException("Detect AirGap installation directory is null.");
+        }
+
+        Collection<File> foundJars = FileUtils.listFiles(new File(baseDir), new PrefixFileFilter(DETECT_JAR_PREFIX), null);
```

Comment:
I'm updating this to filter on both prefix and suffix. It is entirely possible we will be required in the future to include a readme file that has the same name as the jar, with just a different suffix.